ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/deduplication/CommandDeduplicationBase.scala

```diff
@@ -341,12 +370,26 @@ private[testtool] abstract class CommandDeduplicationBase(
   )(request: SubmitRequest, parties: Party*)(implicit
       ec: ExecutionContext
   ): Future[Unit] = {
-    if (deduplicationFeatures.participantDeduplication) {
+    if (deduplicationFeatures.participantDeduplication)
       submitRequestAndAssertSyncDeduplication(ledger, request)
-    } else {
+    else
       submitRequestAndAssertAsyncDeduplication(ledger)(request, parties: _*)
         .map(_ => ())
-    }
+  }
+
+  def submitRequestAndGetStatusCode(
+      ledger: ParticipantTestContext
+  )(request: SubmitRequest, parties: Party*)(implicit ec: ExecutionContext): Future[Code] = {
+    if (deduplicationFeatures.participantDeduplication)
```

Comment:
Such a strict distinction is problematic for Canton. Depending on the timing, Canton will output the rejections synchronously (with status code `ABORTED`) or asynchronously (with status code `ALREADY_EXISTS` in case the completion event for an earlier submission has already been output when the later submission hits deduplication).

Reply:
Ok, in this case we need to change the logic a bit.
If it's OK on the sync response, we check for the completion status, if it's something else then we return that status directly as there will be no completion.

Reply:
That sounds good.